internal/model/vpnoptions_test.go

```diff
@@ -0,0 +1,829 @@
+		},
+		{
+			name:    "should fail with 2-element array",
+			args:    args{[]string{"foo", "bar"}, &OpenVPNOptions{}},
+			wantErr: ErrBadConfig,
+		},
+		{
+			name:    "should fail with lowercase option",
+			args:    args{[]string{"sha1"}, &OpenVPNOptions{}},
+			wantErr: ErrBadConfig,
+		},
+		{
+			name:    "should fail with unknown option",
+			args:    args{[]string{"SHA666"}, &OpenVPNOptions{}},
+			wantErr: ErrBadConfig,
+		},
+		{
+			name:    "should not fail with good option",
+			args:    args{[]string{"SHA512"}, &OpenVPNOptions{}},
+			wantErr: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if _, err := parseAuth(tt.args.p, tt.args.o); !errors.Is(err, tt.wantErr) {
+				t.Errorf("parseAuth() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+		})
+	}
+}
+
+func Test_parseAuthUser(t *testing.T) {
+	makeCreds := func(credStr string) string {
+		f, err := os.CreateTemp(t.TempDir(), "tmpfile-")
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err := f.Write([]byte(credStr)); err != nil {
+			t.Fatal(err)
+		}
+		return f.Name()
+	}
+
+	baseDir := func() string {
+		return os.TempDir()
+	}
+
+	type args struct {
+		p []string
+		o *OpenVPNOptions
+		d string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr error
+	}{
+		{
+			name: "parse good auth",
+			args: args{
+				p: []string{makeCreds("foo\nbar\n")},
+				o: &OpenVPNOptions{},
+				d: baseDir(),
+			},
+			wantErr: nil,
+		},
+		{
+			name: "path traversal should fail",
+			args: args{
+				p: []string{"/tmp/../etc/passwd"},
+				o: &OpenVPNOptions{},
+				d: baseDir(),
+			},
+			wantErr: ErrBadConfig,
+		},
+		{
+			name: "parse empty file should fail",
+			args: args{
+				p: []string{""},
+				o: &OpenVPNOptions{},
+				d: baseDir(),
+			},
+			wantErr: ErrBadConfig,
+		},
+		{
+			name: "parse empty parts should fail",
+			args: args{
+				p: []string{},
+				o: &OpenVPNOptions{},
+				d: baseDir(),
+			},
+			wantErr: ErrBadConfig,
+		},
+		{
+			name: "parse less than two lines should fail",
+			args: args{
+				p: []string{makeCreds("foo\n")},
+				o: &OpenVPNOptions{},
+				d: baseDir(),
+			},
+			wantErr: ErrBadConfig,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if _, err := parseAuthUser(tt.args.p, tt.args.o, tt.args.d); !errors.Is(err, tt.wantErr) {
+				t.Errorf("parseAuthUser() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+// TODO(ainghazal): return options object so that it's testable too
+func Test_parseTLSVerMax(t *testing.T) {
+	type args struct {
+		p []string
+		o *OpenVPNOptions
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr error
+	}{
+		{
+			name:    "default",
+			args:    args{o: &OpenVPNOptions{}},
+			wantErr: nil,
+		},
+		{
+			name:    "default with good tls opt",
+			args:    args{p: []string{"1.2"}, o: &OpenVPNOptions{}},
+			wantErr: nil,
+		},
+		{
+			// FIXME this case should probably fail
```

Comment:
Can you replace `FIXME` comments with `TODO(ainghazal)` comments?

Reply:
done!